recipes/protobuf/all/conanfile.py

```diff
@@ -38,13 +39,24 @@ def source(self):
         extracted_folder = self.name + "-" + self.version
         os.rename(extracted_folder, self._source_subfolder)
 
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
     def configure(self):
-        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
+        if self.options.shared:
             del self.options.fPIC
+
             compiler_version = Version(self.settings.compiler.version.value)
             if compiler_version < "14":
                 raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
```

Comment:
Should this piece of code be executed on Linux?

Because I saw this on Linux right now :-(
```
$ conan create . protobuf/3.12.4@ -o protobuf:shared=True
Exporting package recipe
protobuf/3.12.4 exports: File 'conandata.yml' found. Exporting it...
protobuf/3.12.4 exports: Copied 1 '.yml' file: conandata.yml
protobuf/3.12.4 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
protobuf/3.12.4 exports_sources: Copied 2 '.patch' files: upstream-issue-7567-no-export-template-define.patch, upstream-pr-7761-cmake-regex-fix.patch
protobuf/3.12.4: A new conanfile.py version was exported
protobuf/3.12.4: Folder: /home/pfarre/.conan/data/protobuf/3.12.4/_/_/export
protobuf/3.12.4: Exported revision: fcab376e471eb6443f1f3fd012065226
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=9
os=Linux
os_build=Linux
[options]
protobuf:shared=True
[build_requires]
[env]

ERROR: protobuf/3.12.4: Invalid configuration: On Windows Protobuf can only be built with Visual Studio 2015 or higher.

```

Reply:
Yup that's a bug.
You can't build a shared protobuf using a current compiler.
(or it's a feature and you'll have to wait until 2024 when gcc 14 gets released :smile: )